### PR TITLE
Simplify build process

### PR DESCRIPTION
**Pull request by Al2Klimov**
The pre-cleanup from #56 on its own. Let's see if it fixes the recent Docker GHA runs in Icinga DB.

CC @yhabteab

### CODE REVIEW

**Comment by Al2Klimov** [comment]
Yes, it fixes that.

https://github.com/Icinga/icingadb/actions/runs/4042583259/jobs/6950460392

Please merge.

**Comment by Al2Klimov** [comment]
This basically re-builds all the things nicely as we did for Icinga 2.

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+    --mount=type=cache,target=/root/.cache/go-build \
+    ["go", "build", "-ldflags", "-s -w", "./cmd/icingadb"]
+
+RUN ["upx", "icingadb"]
 
+RUN ["bzip2", "-k", "schema/mysql/schema.sql"]
+RUN ["bzip2", "-k", "schema/pgsql/schema.sql"]
+
+
+FROM go-upx as entrypoint
```
Comment (julianbrost) [inline Dockerfile:24]: Is there a benefit in using multiple build stages? It's the same base container, so it should be just fine to build Icinga DB and the entry point in the same container.

**Review by julianbrost — COMMENTED** [review]
Isn't this an incompatible change to the action? I don't see a corresponding PR in the Icinga DB repo.

File: Dockerfile (hunk 1 above)
Comment (Al2Klimov) [reply Dockerfile:24]: Yes, caching. Only changed stuff gets re-built.

File: build.bash [hunk 2]
```diff
@@ -12,28 +12,14 @@ EOF
...
+fi
 
-cp -r /entrypoint .
-cp -r /rootfs .
+if ! docker buildx version; then
+	echo '"docker buildx" not found (see https://docs.docker.com/buildx/working-with-buildx/ )' >&2
+	false
+fi
 
-docker build -f /bldctx/Dockerfile -t icinga/icingadb .
```
Comment (Al2Klimov) [inline build.bash:38]: ## One more reason for this

```
+ docker build -f /bldctx/Dockerfile -t icinga/icingaweb2 .
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
```

File: action.yml [hunk 3]
```diff
@@ -1,6 +1,41 @@
...
+        docker tag icinga/icingadb "icinga/icingadb:$TAG"
+        docker push "icinga/icingadb:$TAG"
+
+    - if: github.event_name == 'push'
+      shell: bash
+      run: |
+        TAG='${{ github.ref_name }}'
+        '${{ github.action_path }}/build.bash' .
+        docker tag icinga/icingadb "icinga/icingadb:$TAG"
+        docker push "icinga/icingadb:$TAG"
```
Comment (julianbrost) [inline action.yml]: This will fail for branches with `/` in the name (like it has already shown in the icinga2 repo).

**Review by julianbrost — CHANGES_REQUESTED** [review]

File: action.yml (hunk 3 above)
Comment (Al2Klimov) [reply action.yml]: ## TODO

* [x] Icinga/docker-icinga2#93

**Comment by Al2Klimov** [comment]
Test pipeline: https://github.com/Icinga/icingadb/actions/runs/4232752022/jobs/7352865183

**Review by julianbrost — APPROVED** [review]

**Pull request merged**